### PR TITLE
Adding new Nemesis - TopPartitions

### PR DESCRIPTION
**Pull request by aleksbykov**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [ ] ~~All new and existing unit tests passed (`hydra unit-tests`)~~
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -859,6 +860,94 @@ def disrupt_snapshot_operations(self):
...
+                sampler_data = sampler_data.groupdict()
+                partitions = re.findall(pattern2, out, re.MULTILINE)
+                for v in partitions:
+                    partition.update({v[0]: {'count': v[1], 'margin': v[2]}})
+                sampler_data.update({'partitions': partition})
+                toppartitions[sampler_data.pop('sampler')] = sampler_data
+            return toppartitions
+
+        self._set_current_disruption("ShowTopPartitions")
+        self.target_node.run_nodetool('cfstats')
```
Comment (bentsi) [inline sdcm/nemesis.py]: add a comment that we run cfstats because of the bug in Scylla

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -859,6 +860,94 @@ def disrupt_snapshot_operations(self):
...
+        ks_cf_list = self.loaders.get_non_system_ks_cf_list(self.loaders.nodes[0], self.cluster.nodes[0])
+        top_partition = random.randint(5, 20)
+        samplers = random.choice(['writes', 'reads', 'writes,reads'])
+        capacity = random.randint(100, 1024)
+        ks, cf = random.choice(ks_cf_list).split('.')
+        duration = random.randint(1000, 10000)
+        cmd = "toppartitions {ks} {cf} {duration} -s {capacity} -k {top_partition} -a {samplers}"
+        result = self.target_node.run_nodetool(cmd.format(**locals()))
```
Comment (bentsi) [inline sdcm/nemesis.py]: use `sub_cmd="toppartitions"` and `args="{ks} {cf} {duration} -s {capacity} -k {top_partition} -a {samplers}"`

File: sdcm/nemesis.py [hunk 3]
```diff
@@ -1364,6 +1453,16 @@ def disrupt(self):
...
+class TopPartitions(Nemesis):
+
+    disruptive = False
+    run_with_gemini = True
+
+    @log_time_elapsed_and_status
+    def disrupt(self):
+        self.disrupt_show_toppartitions()
+
+
```
Comment (bentsi) [inline sdcm/nemesis.py]: Add to `ChaosMonkey`.
@roydahan which nemesis group we should add this Nemesis?

File: sdcm/nemesis.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/nemesis.py]: added

Comment (aleksbykov) [reply sdcm/nemesis.py]: Fixed in branch next-3.1

File: sdcm/nemesis.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/nemesis.py]: fixed

File: sdcm/nemesis.py (hunk 3 above)
Comment (bentsi) [reply sdcm/nemesis.py]: discussed with Roy, please add to ChaosMonkey

Comment (aleksbykov) [reply sdcm/nemesis.py]: @bentsi to LimitedChaosMonkey you mean? because in ChaosMonkey it will run any way

Comment (aleksbykov) [reply sdcm/nemesis.py]: Added to LimitedChaosMonkey

**Pull request merged**